frame/cosmwasm/src/lib.rs

```diff
@@ -482,11 +485,18 @@ pub mod pallet {
 			salt: &[u8],
 			admin: Option<AccountIdOf<T>>,
 			label: ContractLabelOf<T>,
+			message: &[u8],
 		) -> Result<(AccountIdOf<T>, ContractInfoOf<T>), Error<T>> {
+			let pretty_message = {
+				let value: serde_json::Value =
+					serde_json::from_slice(message).map_err(|_| Error::<T>::InvalidJsonMessage)?;
+				serde_json::to_vec_pretty(&value).map_err(|_| Error::<T>::InvalidJsonMessage)?
+			};
 			let code_hash = CodeIdToInfo::<T>::get(code_id)
 				.ok_or(Error::<T>::CodeNotFound)?
 				.pristine_code_hash;
```

Comment:
Can we first check if the code is present before hashing the input message please?